mirdata/groove_midi.py

```diff
@@ -0,0 +1,288 @@
+
+        self._data_home = data_home
+        self._track_paths = DATA.index[track_id]
+
+        metadata = DATA.metadata(data_home)
+        if metadata is not None and track_id in metadata:
+            self._track_metadata = metadata[track_id]
+        else:
+            self._track_metadata = {
+                "drummer": None,
+                "session": None,
+                "style": None,
+                "bpm": None,
+                "beat_type": None,
+                "time_signature": None,
+                "midi_filename": None,
+                "audio_filename": None,
+                "duration": None,
+                "split": None,
+            }
+
+        self.drummer = self._track_metadata["drummer"]
+        self.session = self._track_metadata["session"]
+        self.style = self._track_metadata["style"]
+        self.bpm = self._track_metadata["bpm"]
+        self.beat_type = self._track_metadata["beat_type"]
+        self.time_signature = self._track_metadata["time_signature"]
+        self.duration = self._track_metadata["duration"]
+        self.split = self._track_metadata["split"]
+        self.midi_filename = self._track_metadata["midi_filename"]
+        self.audio_filename = self._track_metadata["audio_filename"]
+
+        self.midi_path = os.path.join(self._data_home, self._track_paths["midi"][0])
+
+        if self._track_paths["audio"][0]:
+            self.audio_path = os.path.join(
+                self._data_home, self._track_paths["audio"][0]
+            )
+        else:
+            self.audio_path = None
+
+    @property
+    def audio(self):
+        """(np.ndarray, float): audio signal, sample rate"""
+        if self.audio_path:
+            return load_audio(self.audio_path)
+        else:
+            return (None, None)
+
+    @property
+    def midi(self):
+        """(obj): prettyMIDI obj"""
+        return load_midi(self.midi_path)
+
+
+def load_audio(audio_path):
+    """Load a Groove MIDI audio file.
+
+    Args:
+        audio_path (str): path to audio file
+
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+
+    """
+    return librosa.load(audio_path, sr=22050, mono=True)
+
+
+def load_midi(midi_path):
+    """Load a Groove MIDI midi file.
+
+    Args:
+        midi_path (str): path to midi file
+
+    Returns:
+        midi_data (obj): prettyMIDI object containing MIDI data. Refer to http://craffel.github.io/pretty-midi/ for more details. 
+
+    """
+    return pretty_midi.PrettyMIDI(midi_path)
+
+
+def download(data_home=None):
+    """Download Groove MIDI.
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    download_utils.downloader(
+        data_home, downloads=[AUDIO_MIDI_REMOTE], cleanup=True,
```

Comment:
I'm finding a bug here:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-f95fc2f19ea0> in <module>
----> 1 gm.download(data_home="/beegfs/vl1019/mir_datasets")

~/mirdata/mirdata/groove_midi.py in download(data_home)
    216 
    217     download_utils.downloader(
--> 218         data_home, downloads=[AUDIO_MIDI_REMOTE], cleanup=True,
    219     )
    220 
```

@kyungyunlee I think you meant `zip_downloads`?

Reply:
Oops that got almost missed. Thanks!